components/molecules/DropDown.js

```diff
@@ -2,10 +2,14 @@ import styles from '../../styles/DropDown.module.css';
 
 export default function DropDown(props) {
 
+    function updateState(ev) {
+        props.state.set(ev.target.value);
+    }
+
     return (
         <div className={styles.container}>
             <label htmlFor={props.id}>{props.label}</label>
-            <select name={props.id} id={props.id} defaultValue={props.state.get}>
+            <select name={props.id} id={props.id} defaultValue={props.state.get} onChange={updateState}>
```

Comment:
Would this not be more of a bug with your `getResults` function where it's not grabbing the value from the selector but instead just continuing to pass the state? (I get that this would fix it, just wondering if it's the best solution) https://github.com/DTS-STN/Personalizer/blob/456421ca3a8885183e5b10fcd5e71c562e9f9c4e/components/templates/Personalizer.js#L25

Reply:
No, this would be the safer way to do it. In React you want to interact with the DOM as little as possible and pass everything as state. State can flow both forwards and backwards, so you know when you use a state variable it is always updated.